### PR TITLE
PVS is now aware of entity anchoring.

### PR DESCRIPTION
**Pull request by Acruid**
PVS is now aware of entity anchoring. This allows large PVS perf improvements.

Sick Gains Brah
*30TPS server with 1 Client running around the station for 60 seconds, opening hallway doors.*

Here is the old profiler performance for comparison. https://github.com/space-wizards/RobustToolbox/pull/1629
the old code could fit (15,000 / 1,902 = ~7) clients per thread, new one can fit (15,000 / 344 = ~44) clients per thread with each thread taking 25% of CPU time.
Roughly 44 clients per thread, a decent 8 core CPU in can easily process 300 players. I'm benchmarking this on an old I5-4670, so perf is going to be better on newer server hardware.

### CODE REVIEW

File: Robust.Server/GameStates/EntityViewCulling.cs [hunk 1]
```diff
@@ -159,22 +158,89 @@ private HashSet<EntityUid> GetSessionViewers(ICommonSession session)
...
+                        // for each grid, calculate the chunks that are inside the box
+                        foreach (var chunk in grid.GetMapChunks().Values)
+                        {
+                            // for each chunk, check dirty
+                            if (chunk.LastAnchoredModifiedTick < fromTick)
+                                continue;
+
+                            // for each chunk, check actually intersects view
+                            if (!chunk.CalcWorldBounds().Intersects(in viewBox))
+                                continue;
```
Comment (metalgearsloth) [inline Robust.Server/GameStates/EntityViewCulling.cs]: Would be good to potentially just use their origins instead in the future as this has to recalculate the grid's worldposition every time.

File: Robust.Server/GameStates/EntityViewCulling.cs [hunk 2]
```diff
@@ -235,7 +310,7 @@ private List<EntityState> GenerateEntityStates(ICommonSession session, GameTick
 
                     // skip sending anchored entities (walls)
                     var xform = _compMan.GetComponent<ITransformComponent>(entityUid);
-                    if (xform.Anchored && _entMan.GetEntity(entityUid).LastModifiedTick <= lastMapUpdate)
+                    if (xform.Anchored && xform.Owner.LastModifiedTick <= lastMapUpdate)
```
Comment (metalgearsloth) [inline Robust.Server/GameStates/EntityViewCulling.cs]: Should this be Anchored || lastmodified now? As the anchored state is already added in the chunk part right?

**Review by PJB3005 — APPROVED** [review]

**Comment by Acruid** [comment]
This is the network update portion of https://github.com/space-wizards/RobustToolbox/issues/1719.

**Comment by PJB3005** [comment]
Any response on the review comments by metalgearsloth @Acruid?

File: Robust.Server/GameStates/EntityViewCulling.cs (hunk 2 above)
Comment (Acruid) [reply Robust.Server/GameStates/EntityViewCulling.cs]: After looking at this further, there is actually a nasty edge case in this whole section of code, where a mover in PVS is parented to an anchored ent outside PVS on a non-visible chunk. All of this code needs to be redesigned.

**Comment by metalgearsloth** [comment]
You still working on this @Acruid ?

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**